### DISCUSSION TITLE
dotnet8 and GitVersion produce different results depending on the Nuget version and running environment

### DISCUSSION THREAD

**Discussion by soty** [GitTools/GitVersion#4205] [Q&A]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.MsBuild, AzureDevops task

### GitVersion version

5.1.2 and 6.0.2

### Operating system

Windows

### What are you seeing?

Hi! I have recently migrated from the full dotnet framework to dotnet8. With this migration I had to update most of the Nuget packages.. And here is where my problem arose. 
I installed dotnet-gitversion and this is working well on my local machine, generating a fully descriptive output like this 

> PM> dotnet-gitversion
{
  "AssemblySemFileVer": "3.11.0.0",
  "AssemblySemVer": "3.11.0.0",
  "BranchName": "feature/10123-common-library-migration",
  "BuildMetaData": 1618,
  "CommitDate": "2024-09-09",
  "CommitsSinceVersionSource": 1618,
  "EscapedBranchName": "feature-10123-common-library-migration",
  "FullBuildMetaData": "1618.Branch.feature-10123-common-library-migration.Sha.bc7635ebd6c3bc0c7badd7a555b85e233af5b794",
  "FullSemVer": "3.11.0-10123-common-library-migration.1+1618",
  "InformationalVersion": "3.11.0-10123-common-library-migration.1+1618.Branch.feature-10123-common-library-migration.Sha.bc7635ebd6c3bc0c7badd7a555b85e233af5b794",
  "Major": 3,
  "MajorMinorPatch": "3.11.0",
  "Minor": 11,
  "Patch": 0,
  "PreReleaseLabel": "10123-common-library-migration",
  "PreReleaseLabelWithDash": "-10123-common-library-migration",
  "PreReleaseNumber": 1,
  "PreReleaseTag": "10123-common-library-migration.1",
  "PreReleaseTagWithDash": "-10123-common-library-migration.1",
  "SemVer": "3.11.0-10123-common-library-migration.1",
  "Sha": "bc7635ebd6c3bc0c7badd7a555b85e233af5b794",
  "ShortSha": "bc7635e",
  "UncommittedChanges": 2,
  "VersionSourceSha": "9db83d5e7a6604b3b6f0d8f4cea6b9d53a1553a1",
  "WeightedPreReleaseNumber": 30001
}

Yet when building my app, my binaries are being signed with only this 
![image](https://github.com/user-attachments/assets/54f5a170-99a1-41f9-9b66-2640004c40ee)

like the entire InformationVersion is being replaced with a default 1.0.0+[Sha] where I am expecting the entire thing to be there. 

When I updated to GitVersion.MsBuild 5.12.0, I am getting back my information - happy days:
![image](https://github.com/user-attachments/assets/37230f81-b459-4e00-bbc0-634113a3f66c)

But then when I commit my code and want to build it with AzureDevops pipelines, regardless if I use a pipeline with or without GitVersion Task I am getting this error 
`net8.0\AssemblyInfo.g.cs(12,28): Error CS7034: The specified version string '20220224.1.0.0' does not conform to the required format - major[.minor[.build[.revision]]]
`
this is also my .csproj config as I was checking different flags but nothing was working for me like I wished to

>   <PropertyGroup>
    
    
    
    
    
    <IncludeSourceRevisionInInformationalVersion>true</IncludeSourceRevisionInInformationalVersion>
    
    <UseFullSemVerForNuGet>true</UseFullSemVerForNuGet>
    <Deterministic>false</Deterministic>
  </PropertyGroup>

### What is expected?

I would like results including the full InformationalVersion piece being embedded to my DLL when using the Nuget package on both my local env as well as my CI hosted on Azure Devops. I am not using any fancy configs, reading tags or any of this. Just my recent branch name and commit hash that can me parsed from that string.

### Steps to Reproduce

I installed this Nuget on two projects: 
1) migrated c# class library from dotnet 4.8 to dotnet 8
2) a brand new webAPI project in dotnet8

Install dotnet-gitversion
Install GitVersion.MsBuild 5.12.0
build solution -> a full info is embedded to a binary
commit to repository and run CI build -> error about version not confirming with patters

Update to GitVersion.MsBuild 6.0.2
build solution -> just a commit Sha is available
commit to repository and run CI build -> just a commit Sha is available

**Comment by soty** [comment]
hello, would someone have any input to that please?

**Reply by HHobeck** [reply]
Have you thought about it that this error has nothing to do with GitVersion? The error message comes from the dotnet compiler and says that the version is not in the correct format. Probably the major version is a short number and out of range.

**Reply by HHobeck** [reply]
A simple Console Application in dotnet 8.0:
![image](https://github.com/user-attachments/assets/7f24f05d-32ad-4fda-9996-f9191f108313)

**Reply by HHobeck** [reply]
Please analyse your issue in more detail and simplify your scenario. I'm going to close this issue.

**Reply by soty** [reply]
hey @HHobeck
 I have, but it does not seem to be the ultimate solution here. As you pointed out, the version seems to be too long, yet why would the same config build on my local machine? Besides this is only on the previous tool version. When I migrate to the 6.0.2 version, I am not seeing the full info as expected. Why is that? This does not come from dot net compiler.. 
Thanks

**Reply by HHobeck** [reply]
I don't know why you are facing this error. You need to invest it in more detail. Have you tried to reply it locally step by step by using the same commands like it was executed in the pipeline?
